### ISSUE TITLE
HTTPStatus: not-an-iterable

### ISSUE THREAD

**Issue by huwcbjones** [PyCQA/pylint#7307]
### Bug description

Iterating over HTTPStatus leads to a false positive.

```py
# pylint: disable=missing-module-docstring
from http import HTTPStatus

for i in HTTPStatus:
    print(i)
```

The following examples seem to work fine
```py
class Days(IntEnum):
    MONDAY = 1
    TUESDAY = 2

class Days(IntEnum):
    def __new__(cls, value, abbr=""):
        obj = int.__new__(cls, value)
        obj._value_ = value
        obj.abbr = abbr
        return obj

    MONDAY = 1, "Mon"
    TUESDAY = 2, "Tue"
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************ Module test
test.py:4:9: E1133: Non-iterable value HTTPStatus is used in an iterating context (not-an-iterable)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.1 (default, Mar 15 2021, 18:22:18)
[Clang 12.0.0 (clang-1200.0.32.27)]
```

### OS / Environment

macOS
